src/main/kotlin/ink/abb/pogo/scraper/tasks/ReadGpx.kt

```diff
@@ -0,0 +1,47 @@
+package ink.abb.pogo.scraper.tasks
+
+import com.pokegoapi.google.common.geometry.S2LatLng
+import ink.abb.pogo.scraper.Bot
+import ink.abb.pogo.scraper.Context
+import ink.abb.pogo.scraper.Settings
+import ink.abb.pogo.scraper.Task
+import ink.abb.pogo.scraper.util.Log
+import org.jdom2.Document
+
+import org.jdom2.input.SAXBuilder
+import java.io.File
+
+/**
+ * Created by Peyphour on 8/11/16.
+ */
+
+class ReadGpx : Task {
+    override fun run(bot: Bot, ctx: Context, settings: Settings) {
+
+        var builder: SAXBuilder = SAXBuilder()
+        var document: Document
+        try {
+            document = builder.build(File(settings.gpxFile))
+        } catch(e: Exception) {
+            // Invalid file or no file specified : Expected behaviour
+            return
+        }
+
+        val coords = document.rootElement.getChild("trk", document.rootElement.namespace)
+                        .getChild("trkseg", document.rootElement.namespace)
+
+        var i: Int = settings.gpxRepeat
+
+        Log.green("GPX plugin : reading file " + settings.gpxFile + " with $i repeats")
+
+        // gpxRepeat == 0 or -1 : No coordinates will be added
+        while(i > 0) {
```

Comment:
Could you maybe put this in a task to add the next iteration when the previous ended. That way an "infinite" option is possible in a "clean" way (instead of adding your route eg "99999" times, which can be very memory intensive)